### PR TITLE
job status bugfix

### PR DESCRIPTION
**Pull request by FeLoch**
See the attached issue.

### CODE REVIEW

**Comment by jan-janssen** [comment]
Please add `refresh_status()` before https://github.com/pyiron/pyiron/blob/master/pyiron/base/job/jobstatus.py#L157 rather than changing the job class.

**Comment by FeLoch** [comment]
is done!

**Comment by jan-janssen** [comment]
I would combine both lines: 

```
if [key for key, val in self._status_dict.items() if val][0] in ["running", "collect", "submitted"]:
             if self.database and self.job_id:
```

to: 

```
if self.database and self.job_id and any([self._status_dict[i] for i in ["running", "collect", "submitted"]]):
```

**Comment by jan-janssen** [comment]
Can you add `"created"` it seems like that is required for non modal jobs which causes the tests to fail:
```
if self.database and self.job_id and any([self._status_dict[i] for i in ["running", "collect", "submitted", "created"]]): 
```

File: pyiron/base/job/jobstatus.py [hunk 1]
```diff
@@ -183,7 +184,7 @@ def refresh_status(self):
         Refresh the job status - check if the database and job_id are set and if this is the case load the job status
         from the database.
         """
-        if self.database and self.job_id:
+        if self.database and self.job_id and any([self._status_dict[i] for i in ["running", "collect", "submitted"]]):
```
Comment (jan-janssen) [inline pyiron/base/job/jobstatus.py:187]:
```suggestion
if self.database and self.job_id and any([self._status_dict[i] for i in ["running", "collect", "submitted", "created"]]):
```

**Comment by FeLoch** [comment]
I guess, it would be better to just exclude the really "finished" jobs.

`for i not in ["finished", "aborted", "warning", "not_converged"]`

**Comment by jan-janssen** [comment]
> I guess, it would be better to just exclude the really "finished" jobs.
> 
> `for i not in ["finished", "aborted", "warning", "not_converged"]`

But it is a for loop ...

**Comment by FeLoch** [comment]
Oh shit, I should take a break ...

File: pyiron/base/job/jobstatus.py (hunk 1 above)
Comment (FeLoch) [reply pyiron/base/job/jobstatus.py]: nice!

**Comment by samwaseda** [comment]
It would be great if you could remind me of the following question at the pyiron meeting if I forget it:

Is it feasible to update the `timestop` tag in the job table if the status is to be updated? I think it would good enough if it's set to the time point when the refreshing is performed.

**Pull request merged**